doc/api/stream.md

```diff
@@ -1667,6 +1667,13 @@ of a stream that are intended for use by consumers (as described in the
 [API for Stream Consumers][] section). Doing so may lead to adverse side effects
 in application code consuming the stream.
 
+It is highly discouraged to override any public method or to emit any internal
+events such as `'error'`, `'data'`, `'end'`, `'finish'` and `'close'` manually
+through `.emit(eventName)` instead of using API provided callbacks or
+`.destroy(err)`. Doing so can break current and future stream invariants
```

Comment:
Maybe: `instead of using API calls and API provided callbacks`.

Reply:
I just removed that part. It just makes things confusing.

Reply:
Ok then.